app/controllers/posts_controller.rb

```diff
@@ -346,25 +346,23 @@ def mark_hidden
   end
 
   def change_status
-    begin
-      new_status = Post::Status.get_status(params[:status].upcase)
-    rescue NameError
+    unless Post.statuses.keys.include?(params[:status])
       flash[:error] = "Invalid status selected."
-    else
-      @post.status = new_status
-      begin
-        Post.transaction do
-          @post.save!
-          @post.mark_read(current_user, @post.tagged_at)
-        end
-      rescue ActiveRecord::RecordInvalid
-        flash[:error] = {
-          message: "Status could not be updated.",
-          array: @post.errors.full_messages
-        }
-      else
-        flash[:success] = "Post has been marked #{params[:status]}."
+      return redirect_to post_path(@post)
+    end
+
+    begin
+      Post.transaction do
+        @post.update!(status: params[:status])
+        @post.mark_read(current_user, @post.tagged_at)
       end
+    rescue ActiveRecord::RecordInvalid
+      flash[:error] = {
+        message: "Status could not be updated.",
+        array: @post.errors.full_messages
+      }
+    else
+      flash[:success] = "Post has been marked #{@post.status}."
```

Comment:
Only changed this because I (incorrectly) thought we were submitting integers here instead of strings. Makes literally no difference, they're the same.

Reply:
I had a PR to submit ints instead of strings but you went 'um this seems suboptimal'. that might be what you were thinking of, though.